gene2phenotype_project/gene2phenotype_app/views.py

```diff
@@ -718,57 +724,83 @@ def post(self, request, stable_id):
 
         return response
 
-
-"""
-    Add a new curation entry.
-    It is only available for authenticated users.
-"""
 class AddCurationData(BaseAdd):
+    """
+        Add a new curation entry.
+        It is only available for authenticated users.
+        We do not need to check for authenticated users because of the user management issues.
+    """
     serializer_class = CurationDataSerializer
-    permission_classes = [permissions.IsAuthenticated]
-
+    
     def post(self, request):
-        user = self.request.user
-
-        if not user.is_authenticated:
-            return Response({"message": "Permission denied"}, status=status.HTTP_403_FORBIDDEN)
+        """
+            Handle POST requests.
 
-        serializer_class = CurationDataSerializer(data=request.data, context={"user": user})
-        if serializer_class.is_valid():
+            Args:
+                request: The HTTP request object.
 
-            # TODO: validate data here
-            # - does JSON have the correct format?
-            # - are the values valid?
-            # - user has permission to edit the panel (if panel available yet)?
-            # if session_name is defined - is it already being used
+            Returns:
+                A Response object with appropriate status and message.
+        """
+        json_file_path = settings.BASE_DIR.joinpath("gene2phenotype_app", "utils", "curation_schema.json")
+        try:
+            with open(json_file_path, 'r') as file:
+                schema = json.load(file)
+        except FileNotFoundError:
+            return Response({"message": "Schema file not found"}, status=status.HTTP_404_NOT_FOUND)
 
-            serializer_class.save()
-            response = Response({"message": f"Data saved successfully"}, status=status.HTTP_200_OK)
+        # Validate the JSON data against the schema
+        try:
+            validate(instance=request.data, schema=schema)
+        except exceptions.ValidationError as e:
+            return Response({"message": "JSON data does not follow the required format, Required format is" + str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+        serializer = self.serializer_class(data=request.data)
+        if serializer.is_valid():
+            
+            serializer.save()
+            return Response({"message": "Data saved successfully"}, status=status.HTTP_200_OK)
         else:
-            error_message = serializer_class.errors.get('locus', 'Problem saving the data')
-            response = Response({"message": error_message}, status=status.HTTP_400_BAD_REQUEST)
-
-        return response
+            return Response({"message": "Problem saving the data" + str(serializer.errors) , "errors": serializer.errors}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
This is printing the message twice. It is enough to only print the last part `"errors": serializer.errors`.
Example:
```
{
    "message": "Problem saving the data{'message': [ErrorDetail(string=\"Data already under curation. Please check session 'G2P03746'\", code='invalid')]}",
    "errors": {
        "message": [
            "Data already under curation. Please check session 'G2P03746'"
        ]
    }
}
```

Reply:
Hi @seeta-ramaraju, i think this might affect how you call it in the frontend. Just wanted you to note this. Thank you